### PR TITLE
Consistently spell "datatype" as one word.

### PR DESCRIPTION
**Pull request by leifwickland**
The vast majority of usages are as one word. While "datatype" is a
nonstandard compound word, at least it's be consistent.

### CODE REVIEW

**Comment by eed3si9n** [comment]
Maybe I picked it up from Gibbons - http://www.cs.ox.ac.uk/research/pdt/ap/dgp/
I generally like single concept to be a compound word, like _typeclass_ instead of _type class_.

**Comment by eed3si9n** [comment]
<blockquote class="twitter-tweet" lang="en"><p lang="en" dir="ltr">Language evolves.Today we change AP style from e-mail to email, no hyphen. Our editors will announce it at <a href="https://twitter.com/hashtag/ACES2011?src=hash">#ACES2011</a> today.</p>&mdash; AP Stylebook (@APStylebook) <a href="https://twitter.com/APStylebook/status/48798366980780033">March 18, 2011</a></blockquote>

<script async src="//platform.twitter.com/widgets.js" charset="utf-8"></script>

**Comment by leifwickland** [comment]
I understand the impulse to have a single word for a single concept; I always want to write "typeclass."  I mentioned that the spelling was nonstandard only because I was proposing a change to a technically incorrect spelling.

File: docs/08/d.md [hunk 1]
```diff
@@ -114,7 +114,7 @@ type Option[+A] = Free[Trivial, A]
 There's also iteratees implementation based on free monads.
 Finally, he summarizes free monads in nice bullet points:
 
-> - A model for any recursive data type with data at the leaves.
+> - A model for any recursive datatype with data at the leaves.
```
Comment (eed3si9n) [inline docs/08/d.md]: This is from I think Rúnar's slide. I'm totally not going to do the whole [sic] thing.

**Comment by leifwickland** [comment]
Good point about modifying Runar's code. I blinding replaced. I'll unfix that.

**Pull request merged**